File: tests/test_code/js/chained/chained.js
class Chain {
    constructor(val) {
        this.val = val;
    }

    add(b) {
        this.val += b
        return this;
    }

    sub(b) {
        this.val -= b;
        return this;
    }

    mul(b) {
        this.val *= b;
        return this;
    }
}

console.log((new Chain(5)).add(5).sub(2).mul(10));

File: tests/test_code/py/chained/chained.py
class Chain():
    def __init__(self, val):
        self.val = val

    def add(self, b):
        self.val += b
        return self

    def sub(self, b):
        self.val -= b
        return self

    def mul(self, b):
        self.val *= b
        return self


print(Chain(5).add(5).sub(2).mul(10))
